frame/aura/src/lib.rs

```diff
@@ -180,10 +202,19 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 	{
 		// instant changes
 		if changed {
-			let next_authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
+			let next_authorities = validators
+				.map(|(_, k)| k)
+				// Only take a number of authorities that would fit in the set.
+				.take(T::MaxAuthorities::get() as usize)
+				.collect::<Vec<_>>();
+			let bounded_next_authorities = BoundedVec::<T::AuthorityId, T::MaxAuthorities>::force_from(
+				next_authorities,
+				Some("Aura New Session"),
+			);
```

Comment:
```suggestion
			let next_authorities = validators
				.map(|(_, k)| k)
				.collect::<Vec<_>>();
			let bounded_next_authorities = BoundedVec::<T::AuthorityId, T::MaxAuthorities>::force_from(
				next_authorities,
				Some("Aura New Session"),
			);
```

Now you at least to the warn via `force_from`.

Reply:
If you remove the take though, then the storage may actually be bigger, no?

The idea is that we do not want to allow storage to get bigger. Would be happy to have a `truncating_into`

Reply:
yeah, I added what you need to this branch as `truncating_from` (just to be more aligned with `unchecked_from` and `force_from`).